python/TestHarness/schedulers/Scheduler.py

```diff
@@ -102,18 +102,45 @@ def killRemaining(self):
             tester_data.killProcess()
         self.job_queue_count = 0
 
-    def run(self, job_container):
+    def reportSkipped(self, jobs):
+        """
+        Allow derived schedulers to do something with skipped jobs
+        """
+        return
+
+    def preLaunch(self, job_dag):
+        """
+        Allow derived schedulers to modify the DAG before jobs are launched
+        """
+        return
+
+    def run(self, job):
         """ Call derived run method """
         return
 
-    def postRun(self, job_container):
-        """ Call derived postRun method """
+    def postRun(self, job):
+        """
+        Allow derived schdulers to perform post run methods on job
+        """
         return
 
     def cleanUp(self):
         """ Allow derived schedulers to perform cleanup operations """
         return
 
+    def notifyFinishedSchedulers(self):
+        """ Notify derived schedulers we are finished """
+        return
+
+    def adjustJobStatus(self, job, status, caveat):
```

Comment:
I'm not sure I understand why this method exists. There must be a setJobStatus() somewhere? This one does the same thing but locks? Is the caveat different too? Just trying to understand the usage of the two.

Reply:
I can name it setJobStatus now that you mention it. I like that name better. I named it adjust, because there is nothing stopping a tester from setting their own status themselves. But maybe that is a poor idea.

Reply:
👍